src/components/BootswatchThemeSelect.vue

```diff
@@ -0,0 +1,116 @@
+<template>
+  <div>
+    <!-- <img :src="theme.thumbnail" style="height: 30px;" /> -->
+    <b-nav-item-dropdown
+      id="my-nav-dropdown"
+      text="Theme"
+      toggle-class="nav-link-custom"
+      @show.once="fetchApi"
+      right
+      lazy
+    >
+      <b-dropdown-item v-if="themes.length === 0">
+        <b-spinner small></b-spinner>
+        Loading Themes...
+      </b-dropdown-item>
+
+      <!-- TODO Add  v-b-tooltip.hover.right=="{ variant: 'info' }" for tooltip class rendered from bootswatch-->
+      <b-dropdown-item-button
+        v-for="(theme, key) in themes"
+        :key="key"
+        @click="handleClick"
+        :value="theme.cssCdn"
+        :active="activeTheme === theme.name"
+        v-b-tooltip.hover.right
+        :title="theme.description"
+        >{{ theme.name }}
+      </b-dropdown-item-button>
+    </b-nav-item-dropdown>
+  </div>
+</template>
+
+<script>
+import axios from 'axios';
+
+export default {
+  data() {
+    return {
+      activeTheme: '',
+      themes: [],
+      supportsLocalStorage: false,
+    };
+  },
+  name: 'BootswatchThemeSelect',
+  created() {
+    // this.fetchApi();
+    // We don't need to call the api here because it will be called once the show event is triggered from clicking on the dropdown.
+  },
+  mounted() {
+    this.supportsLocalStorage = window.localStorage.length > 0;
```

Comment:
I don't think this is needed (https://caniuse.com/#search=localstorage)

All browsers since IE8 seem to support this.
Now IE8 IS still around - but even my (VERY SLOWLY moving company updated to IE11 as default now :partying_face: )

Reply:
Jeez yea, good point. I was sort of following along with an outdated tutorial from 2013.